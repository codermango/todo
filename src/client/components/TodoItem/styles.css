.todoItem {
  height: 50px;
  padding: 0 10px;
  display: flex;
  align-items: center;
  justify-content: space-between;
  border: 1px solid #ededed;
  cursor: all-scroll;
}

.todoItem:hover {
  background-color: gray;
}

.text {
  flex: 1;
  padding: 0 10px;
}
