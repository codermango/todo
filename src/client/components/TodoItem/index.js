import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import DeleteButton from '../DeleteButton';
import Checkbox from '../Checkbox';
import { deleteTodo, toggleTodo } from '../../actions';

import styles from './styles.css';

class TodoItem extends Component {

  static propTypes = {
    deleteTodo: PropTypes.func.isRequired,
    toggleTodo: PropTypes.func.isRequired,
    data: PropTypes.shape({
      text: PropTypes.string.isRequired,
      done: PropTypes.bool.isRequired
    }).isRequired
  }

  handleDone = () => {
    const { data = {}, toggleTodo } = this.props;
    toggleTodo({ id: data.id });
  }

  handleDeleteTodo = () => {
    const { data = {}, deleteTodo } = this.props;
    deleteTodo({ id: data.id });
  }

  render() {
    const {
      data = {},
      draggable,
      onDragStart,
      onDragOver
    } = this.props;
    return (
      <div
        className={styles.todoItem}
        draggable={draggable}
        onDragStart={onDragStart}
        onDragOver={onDragOver}
        data-test-id="todo-item"
      >
        <Checkbox checked={data.done} onDone={this.handleDone} />
        <span className={styles.text}>{data.text}</span>
        <DeleteButton onDelete={this.handleDeleteTodo} />
      </div>
    )
  }
}

const mapDispatchToProps = dispatch => ({
  deleteTodo: ({ id }) => dispatch(deleteTodo({ id })),
  toggleTodo: ({ id }) => dispatch(toggleTodo({ id }))
});

export default connect(null, mapDispatchToProps)(TodoItem);
