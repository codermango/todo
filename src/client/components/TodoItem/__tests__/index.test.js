import React from 'react';
import { shallow } from 'enzyme';
import TodoItem from '../index';


describe('<TodoItem />', () => {

  const props = {
    data: { id: 'id1', text: 'text1', done: false },
    toggleTodo: jest.fn(),
    deleteTodo: jest.fn()
  };

  it('renders correctly', () => {
    const wrapper = shallow(<TodoItem {...props} />);
    expect(wrapper.instance()).toMatchSnapshot();
  });

  it('handleDone works correctly', () => {
    const wrapper = shallow(<TodoItem {...props} />);
    const instance = wrapper.instance();
    instance.handleDone();
    expect(instance.props.toggleTodo).toHaveBeenCalledTimes(1);
  });

  it('handleDeleteTodo works correctly', () => {
    const wrapper = shallow(<TodoItem {...props} />);
    const instance = wrapper.instance();
    instance.handleDeleteTodo();
    expect(instance.props.deleteTodo).toHaveBeenCalledTimes(1);
  });
});
