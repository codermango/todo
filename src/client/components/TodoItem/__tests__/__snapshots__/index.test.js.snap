// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TodoItem /> renders correctly 1`] = `
TodoItem {
  "context": Object {},
  "handleDeleteTodo": [Function],
  "handleDone": [Function],
  "props": Object {
    "data": Object {
      "done": false,
      "id": "id1",
      "text": "text1",
    },
    "deleteTodo": [MockFunction],
    "toggleTodo": [MockFunction],
  },
  "refs": Object {},
  "setState": [Function],
  "state": null,
  "updater": Updater {
    "_callbacks": Array [],
    "_renderer": ReactShallowRenderer {
      "_context": Object {},
      "_didScheduleRenderPhaseUpdate": false,
      "_dispatcher": Object {
        "readContext": [Function],
        "useCallback": [Function],
        "useContext": [Function],
        "useDebugValue": [Function],
        "useEffect": [Function],
        "useImperativeHandle": [Function],
        "useLayoutEffect": [Function],
        "useMemo": [Function],
        "useReducer": [Function],
        "useRef": [Function],
        "useState": [Function],
      },
      "_element": <TodoItem
        data={
          Object {
            "done": false,
            "id": "id1",
            "text": "text1",
          }
        }
        deleteTodo={[MockFunction]}
        toggleTodo={[MockFunction]}
      />,
      "_firstWorkInProgressHook": null,
      "_forcedUpdate": false,
      "_instance": [Circular],
      "_isReRender": false,
      "_newState": null,
      "_numberOfReRenders": 0,
      "_renderPhaseUpdates": null,
      "_rendered": <div
        className="todoItem"
        data-test-id="todo-item"
      >
        <Checkbox
          checked={false}
          onDone={[Function]}
        />
        <span
          className="text"
        >
          text1
        </span>
        <DeleteButton
          onDelete={[Function]}
        />
      </div>,
      "_rendering": false,
      "_updater": [Circular],
      "_workInProgressHook": null,
    },
  },
}
`;
